lib/gui/pages/settings/controllers/settings.js

```diff
@@ -53,34 +54,45 @@ module.exports = function(WarningModalService, SettingsModel) {
   this.model = SettingsModel;
 
   /**
-   * @summary Enable a dangerous setting
+   * @summary Toggle setting
    * @function
    * @public
    *
-   * @param {String} name - setting name
-   * @param {Object} options - options
-   * @param {String} options.description - modal description
-   * @param {String} options.confirmationLabel - modal confirmation label
+   * @description
+   * If warningOptions is given, it should be an object having `description` and `confirmationLabel`;
+   * these will be used to present a user confirmation modal before enabling the setting.
+   *
+   * @param {String} setting - setting key
+   * @param {Object} warningOptions - (optional) WarningModalService options
```

Comment:
Maybe worth elaborating on the options instead of simply referring to `WarningModalService`? 

For example:

```
@param {Object} [options] - options
@param {String} [options.description] - warning modal description
@param {String} [options.confirmationLabel] - warning modal confirmation label
```